api/src/main/java/org/openmrs/Allergen.java

```diff
@@ -9,17 +9,30 @@
  */
 package org.openmrs;
 
+import javax.persistence.Column;
+import javax.persistence.Embeddable;
+import javax.persistence.EnumType;
+import javax.persistence.Enumerated;
+import javax.persistence.JoinColumn;
+import javax.persistence.ManyToOne;
+
 import org.apache.commons.lang3.StringUtils;
 
 /**
  * Represent allergen
  */
+@Embeddable
 public class Allergen {
 	
+	@Enumerated(EnumType.STRING)
+	@Column(name = "allergen_type")
 	private AllergenType allergenType;
 	
+	@ManyToOne
```

Comment:
Keep this as not nullable like in the hbm file

Reply:
@wluyima i have fixed it

Reply:
I prefer to set optional = false for the @ManyToOne rather than nullable on JoinColumn because then the constraint can be enforced on the object model itself rather than only in the db

Reply:
okay @wluyima true...

i have fixed it